js/actions/CategoriesActionCreator.js

```diff
@@ -0,0 +1,21 @@
+var api = require('../api.js')
+var RepcoinAppDispatcher = require('../dispatcher/RepcoinAppDispatcher.js');
+var RepcoinConstants = require('../constants/RepcoinConstants.js');
+var ActionTypes = RepcoinConstants.ActionTypes;
+
+module.exports = {
+  getCategories: function() {
+    RepcoinAppDispatcher.handleViewAction({
+      type: ActionTypes.GET_CATEGORIES
+    });
+
```

Comment:
This is how we fetch data from the server in our new Flux architecture. First, we emit an event that will let everyone know that we're about to retrieve data from our server. Then, we call our api and ask it to retrieve data. This api will, once it's recieved the data, create it's own action indicating the data's been received.